src/login/login.component.tsx

```diff
@@ -197,7 +208,11 @@ export default function Login(props: LoginProps) {
 
 function navigate(props, urlConfig: UrlConfig) {
   if (urlConfig.spa) {
-    props.history.push(urlConfig.url);
+    props.history.push(
+      props.location && props.location.state && props.location.state.referrer
```

Comment:
This causes the referrer redirect to only work if the configured loginSuccess redirect page is a SPA page. This logic should be totally independent. I would leave this function as it was and instead embed the logic about `location.state.referrer` into your code above.